DigitalLearningSolutions.Web/Controllers/Register/RegisterAtNewCentreController.cs

```diff
@@ -266,31 +271,35 @@ public async Task<IActionResult> SummaryPost()
         }
 
         [HttpGet]
-        public IActionResult Confirmation()
+        public IActionResult Confirmation(
+            string candidateNumber,
+            bool approved,
+            bool userHasAdminAccountAtCentre,
+            int? centreId
+        )
         {
-            var candidateNumber = (string?)TempData.Peek("candidateNumber");
-            var approvedNullable = (bool?)TempData.Peek("approved");
-            var centreIdNullable = (int?)TempData.Peek("centreId");
-            var hasAdminAccountNullable = (bool?)TempData.Peek("userHasAdminAccountAtCentre");
-            TempData.Clear();
-
-            if (candidateNumber == null || approvedNullable == null || centreIdNullable == null ||
-                hasAdminAccountNullable == null)
+            if (centreId == null)
             {
                 return RedirectToAction("Index");
             }
 
-            var hasAdminAccount = (bool)hasAdminAccountNullable;
-            var approved = (bool)approvedNullable;
-            var centreId = (int)centreIdNullable;
+            var userId = User.GetUserIdKnownNotNull();
+
+            var (unverifiedPrimaryEmail, unverifiedCentreEmails) =
+                userService.GetUnverifiedEmailsForUser(userId);
+            var (_, centreName, unverifiedCentreEmail) = unverifiedCentreEmails.First(uce => uce.centreId == centreId);

```

Comment:
*SingleOrDefault - may turn out to be the same as a verified centre email / primary email.

Reply:
Similar on RegisterInternalAdminController

Reply:
changed in both journeys